qiita_pet/handlers/download.py

```diff
@@ -612,6 +612,53 @@ def get(self):
                             '%m%d%y-%H%M%S'))
 
                     self._set_nginx_headers(zip_fn)
+        else:
+            prep_id = int(prep_id)
+            try:
+                prep = PrepTemplate(prep_id)
+            except QiitaDBUnknownIDError:
+                raise HTTPError(422, reason='Prep does not exist')
+            else:
+                public_raw_download = Study(prep.study_id).public_raw_download
+                if prep.status != 'public':
+                    raise HTTPError(404, reason='Prep is not public. If this '
+                                    'is a mistake contact: %s' %
+                                    qiita_config.help_email)
+                elif data == 'raw' and not public_raw_download:
+                    raise HTTPError(422, reason='No raw data access. If this '
+                                    'is a mistake contact: %s'
+                                    % qiita_config.help_email)
+                else:
```

Comment:
This is getting to be a _lot_ of nested else. I'm sure the logic works but it makes it quite hard to read.

Reply:
I agree with you, I think this `class` shows how it has grown over time, which makes it difficult and complex. Thus, decided to rewrite it. Note that I didn't change the tests as those are checking multiple options, parameters and errors.